docs/Consuming.md

```diff
@@ -95,6 +95,25 @@ consumer.run({
 
 In the example above, if the consumer is shutting down in the middle of the batch, the remaining messages won't be resolved and therefore not committed. This way, you can quickly shut down the consumer without losing/skipping any messages.
 
+## <a name="concurrent-processing"></a> Partition-aware concurrency
+
+By default, [`eachMessage`](Consuming.md#each-message) is invoked sequentially for each message in each partition. In order to concurrently process several messages per once, you can increase the `partitionsConsumedConcurrently` option:
+
+```javascript
+consumer.run({
+    partitionsConsumedConcurrently: 3, // Default: 1
+    eachMessage: async ({ topic, partition, message }) => {
+        // This will be called up to 3 times concurrently
+    },
+})
+```
+
+Messages in the same partition are still guaranteed to be processed in order, but messages from multiple partitions can be processed at the same time. If `eachMessage` consists of asynchronous work, such as network requests or other I/O, this can improve performance. If `eachMessage` is entirely synchronous, this will make no difference.
+
+The same thing applies if you are using [`eachBatch`](Consuming.md#each-batch). Given `partitionsConsumedConcurrently > 1`, you will be able to process multiple batches concurrently.
+
+A guideline for setting `partitionsConsumedConcurrently` would be that it should not be larger than the number of partitions consumed. Depending on whether or not your workload is CPU bound, it may also not benefit you to set it to a higher number than the number of logical CPU cores. A recommendation is to start with a low number and measure if increasing leads to higher throughput.
```

Comment:
Can I set it to a high number to always process all partitions in parallel?  Is the behaviour the same for any setting > # consumed partitions ?

Reply:
Yes. You can set it to a number higher than the number of partitions, and the effective result will be the same as if you set it to the number of partitions.